### ISSUE TITLE
seem it can't work on Mac OSX

### ISSUE THREAD

**Issue by siddontang** [cloudwego/netpoll#52]
I try to write a simple test on Mac OSX below, but it can't work. I use `telnet 127.0.0.1 8080`, type `hello world`, but can't get the response. 

```go
package main

import (
	"context"
	"flag"
	"fmt"
	"time"

	"github.com/cloudwego/netpoll"
)

var (
	addr = flag.String("addr", "127.0.0.1:8080", "server address")
)

func onRequest(_ context.Context, conn netpoll.Connection) error {
	println("hello world")

	var reader, writer = conn.Reader(), conn.Writer()

	defer reader.Release()
	// client send "Hello World", size is 11
	buf, err := reader.Next(11)
	if err != nil {
		fmt.Printf("error %v\n", err)
		return err
	}

	alloc, _ := writer.Malloc(len(buf))
	copy(alloc, buf)
	err = writer.Flush()
	if err != nil {
		fmt.Printf("flush error %v\n", err)
		return err
	}

	return nil
}

func onPrepare(conn netpoll.Connection) context.Context {
	println("hello prepare")
	return context.TODO()
}

func main() {
	flag.Parse()
	l, err := netpoll.CreateListener("tcp", *addr)
	if err != nil {
		panic("create netpoll listener failed")
	}
	defer l.Close()

	println("hello event loop")

	loop, err1 := netpoll.NewEventLoop(
		onRequest,
		netpoll.WithReadTimeout(time.Second),
		netpoll.WithOnPrepare(onPrepare),
	)
	if err1 != nil {
		panic("create event loop failed")
	}

	println("begin to serve")
	loop.Serve(l)
}
```

**Comment by betazk** [comment]
I run you test on Mac,and print the result 'hello world',then read timeout.
And analyzed package send from telnet,the length is 13
![image](https://user-images.githubusercontent.com/3760303/134801744-8190afdd-d9d3-4925-aadb-c27c98d3cb6a.png)
Next(11) method will wait  or timeout,alse we can read length first,for example:
len1:=reader.Len()
buf,_:=read.Next(len1)

**Comment by Hchenn** [comment]
Solution: 

Define var netpoll.EventLoop to persist the return value of NewEventLoop.


Reason: 

`loop, err1 := netpoll.NewEventLoop(...`
The `loop` here is not persistently referenced. The runtime GC will mark it incorrectly and reclaim it, causing the `loop` to be empty and the server loses its response.

**Comment by siddontang** [comment]
Em, I don't think this is the reason. I have tested this on Linux and it works well. 

For the above test, I add `println` in `onPrepare` and `onRequest` but find no output.

**Comment by Hchenn** [comment]
> Em, I don't think this is the reason. I have tested this on Linux and it works well.
> 
> For the above test, I add `println` in `onPrepare` and `onRequest` but find no output.

Did you try what I said ?

**Comment by siddontang** [comment]
yes, tried but still can't work, below

```go
	var loop netpoll.EventLoop
	loop, err = netpoll.NewEventLoop(
		onRequest,
		netpoll.WithReadTimeout(time.Second),
		netpoll.WithOnPrepare(onPrepare),
	)
	if err != nil {
		panic("create event loop failed")
	}

	println("begin to serve")
	loop.Serve(l)
```

**Comment by joway** [comment]
@siddontang hi, plz try：

```
var loop netpoll.EventLoop
func main() {
}
```

**Comment by Hchenn** [comment]
> yes, tried but still can't work, below
> 
> ```go
> 	var loop netpoll.EventLoop
> 	loop, err = netpoll.NewEventLoop(
> 		onRequest,
> 		netpoll.WithReadTimeout(time.Second),
> 		netpoll.WithOnPrepare(onPrepare),
> 	)
> 	if err != nil {
> 		panic("create event loop failed")
> 	}
> 
> 	println("begin to serve")
> 	loop.Serve(l)
> ```

Is this a persistent variable ？Please define it outside the function main.

**Comment by Hchenn** [comment]
var loop netpoll.EventLoop

func main() {
	flag.Parse()
	l, err := netpoll.CreateListener("tcp", *addr)
	if err != nil {
		panic("create netpoll listener failed")
	}
	defer l.Close()
	
	println("hello event loop")
	
	loop, err = netpoll.NewEventLoop(
		onRequest,
		netpoll.WithReadTimeout(time.Second),
		netpoll.WithOnPrepare(onPrepare),
	)
	if err != nil {
		panic("create event loop failed")
	}
	
	println("begin to serve")
	loop.Serve(l)
}

**Comment by Hchenn** [comment]
> var loop netpoll.EventLoop
> 
> func main() {
> flag.Parse()
> l, err := netpoll.CreateListener("tcp", *addr)
> if err != nil {
> panic("create netpoll listener failed")
> }
> defer l.Close()
> 
> ```
> println("hello event loop")
> 
> loop, err = netpoll.NewEventLoop(
> 	onRequest,
> 	netpoll.WithReadTimeout(time.Second),
> 	netpoll.WithOnPrepare(onPrepare),
> )
> if err != nil {
> 	panic("create event loop failed")
> }
> 
> println("begin to serve")
> loop.Serve(l)
> ```
> 
> }

**Comment by siddontang** [comment]
em, amazing, can work now...

**Comment by siddontang** [comment]
but your guide https://github.com/cloudwego/netpoll/blob/main/docs/guide/guide_en.md#12-new-eventloop here doesn't mention this.

**Comment by Hchenn** [comment]
> but your guide https://github.com/cloudwego/netpoll/blob/main/docs/guide/guide_en.md#12-new-eventloop here doesn't mention this.

Indeed, I am going to modify it, THANK YOU

**Comment by Hchenn** [comment]
> but your guide https://github.com/cloudwego/netpoll/blob/main/docs/guide/guide_en.md#12-new-eventloop here doesn't mention this.

Fixed by #53 , Thanks for attention ~

**Comment by Anthony-Dong** [comment]
我理解我今天遇到的这个问题和这个也很相似，看样子直接把CPU吃满了，根据上面讲的 https://blog.csdn.net/idwtwt/article/details/102985319 ，应该就是 syscall.Kevent 返回值一直是0导致空转了。这个问题很好复现，就是启动程序等一会GC后就必现了。
![image](https://user-images.githubusercontent.com/52623370/194804588-2a5ed5af-e590-4a0a-a292-83c6c8626ea4.png)
